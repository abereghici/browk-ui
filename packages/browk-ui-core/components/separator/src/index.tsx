import * as React from 'react';
import * as PropTypes from 'prop-types';
import {Box, safelySpreadBoxProps} from '@browk-ui/box';
import type {BoxProps} from '@browk-ui/box';
import type {Space} from '@browk-ui/style-props';
import {isMarginTokenProp} from '@browk-ui/style-props';

type Orientation = 'horizontal' | 'vertical';
export interface SeparatorProps extends React.HTMLAttributes<HTMLHRElement>, Pick<BoxProps, 'element'> {
  id?: never;
  className?: never;
  orientation: Orientation;
  horizontalSpacing?: Space;
  verticalSpacing?: Space;
}

const Separator = React.forwardRef<HTMLHRElement, SeparatorProps>(
  ({element = 'SEPARATOR', orientation, horizontalSpacing, verticalSpacing, ...props}, ref) => {
    return (
      <Box
        {...safelySpreadBoxProps(props)}
        aria-orientation={orientation}
        element={element}
        margin="space0"
        marginBottom={verticalSpacing}
        marginLeft={horizontalSpacing}
        marginRight={horizontalSpacing}
        marginTop={verticalSpacing}
        as="hr"
        borderWidth="borderWidth0"
        borderColor="colorBorder"
        borderStyle="solid"
        borderBottomWidth={orientation === 'horizontal' ? 'borderWidth10' : null}
        borderLeftWidth={orientation === 'vertical' ? 'borderWidth10' : null}
        width={orientation === 'horizontal' ? 'auto' : null}
        height={orientation === 'vertical' ? 'auto' : null}
        ref={ref}
      />
    );
  }
);

Separator.displayName = 'Separator';

Separator.propTypes = {
  element: PropTypes.string,
  orientation: PropTypes.oneOf(['horizontal', 'vertical'] as Orientation[]).isRequired,
  horizontalSpacing: isMarginTokenProp,
  verticalSpacing: isMarginTokenProp,
};

export {Separator};
