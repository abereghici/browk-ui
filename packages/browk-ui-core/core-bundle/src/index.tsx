export * from '@browk-ui/box';
export * from '@browk-ui/separator';
export * from '@browk-ui/text';
