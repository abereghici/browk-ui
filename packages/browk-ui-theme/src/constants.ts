import type {ValueOf} from '@browk-ui/types';

export const ThemeVariants = {
  DEFAULT: 'default',
} as const;

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type ThemeVariants = ValueOf<typeof ThemeVariants>;
