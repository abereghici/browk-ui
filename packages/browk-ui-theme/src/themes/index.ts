import {DefaultTheme} from './default';

export {DefaultTheme};
