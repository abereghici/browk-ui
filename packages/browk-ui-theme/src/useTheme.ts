import * as React from 'react';
import {ThemeContext} from '@browk-ui/styling-library';
import type {ThemeShape} from './types/themes';

const useTheme = (): ThemeShape => {
  const context = React.useContext(ThemeContext) as ThemeShape;
  if (!context) {
    throw new Error('[useHook]: must be used within the @browk-ui/theme provider');
  }
  return context;
};

export {useTheme};
