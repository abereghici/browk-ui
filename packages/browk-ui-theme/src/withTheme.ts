export {withTheme} from '@browk-ui/styling-library';
