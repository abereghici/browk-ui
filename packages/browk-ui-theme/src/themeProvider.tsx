import * as React from 'react';
import {useReducedMotion, Globals as AnimatedGlobals} from '@browk-ui/animation-library';
import {styled, StylingGlobals, ThemeProvider as StyledThemeProvider} from '@browk-ui/styling-library';
import {DefaultTheme} from './themes';
import {browkGlobalStyles} from './styles/global';
import {browkBaseStyles} from './styles/base';
import {browkFonts} from './styles/fonts';
import {ThemeVariants} from './constants';

export const StyledBase = styled.div(browkBaseStyles);

// eslint-disable-next-line @typescript-eslint/ban-types
function getProviderThemeProps(theme: ThemeVariants, customBreakpoints?: string[]): {} {
  switch (theme) {
    case ThemeVariants.DEFAULT:
    default:
      return {
        ...DefaultTheme,
        breakpoints: customBreakpoints || DefaultTheme.breakpoints,
      };
  }
}

export interface ThemeProviderProps {
  customBreakpoints?: string[];
  theme?: ThemeVariants;
  disableAnimations?: boolean;
}

const ThemeProvider: React.FunctionComponent<ThemeProviderProps> = ({
  customBreakpoints,
  theme = ThemeVariants.DEFAULT,
  disableAnimations = false,
  ...props
}) => {
  const prefersReducedMotion = useReducedMotion();
  React.useLayoutEffect(() => {
    AnimatedGlobals.assign({
      skipAnimation: disableAnimations || prefersReducedMotion,
    });
  }, [disableAnimations, prefersReducedMotion]);

  const providerThemeProps = getProviderThemeProps(theme, customBreakpoints);

  return (
    <StyledThemeProvider theme={providerThemeProps}>
      <StylingGlobals styles={browkGlobalStyles({theme: providerThemeProps})} />
      <StylingGlobals styles={browkFonts} />
      <StyledBase {...props} />
    </StyledThemeProvider>
  );
};

export {ThemeProvider};
