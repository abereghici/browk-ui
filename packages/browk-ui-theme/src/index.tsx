import {ThemeProvider, StyledBase} from './themeProvider';
import type {ThemeProviderProps} from './themeProvider';
import {ThemeConsumer} from './themeConsumer';

const Theme = {
  Provider: ThemeProvider,
  Consumer: ThemeConsumer,
};

export {browkBaseStyles} from './styles/base';
export {browkGlobalStyles} from './styles/global';
export {browkFonts} from './styles/fonts';
export {StyledBase};
export type {ThemeProviderProps};
export type {ThemeShape} from './types/themes';
export type {GenericThemeShape} from './types/GenericThemeShape';
export {DefaultTheme} from './themes';
export {withTheme} from './withTheme';
export {useTheme} from './useTheme';
export {ThemeVariants} from './constants';
export {generateThemeFromTokens} from './generateThemeFromTokens';
export {generateTokensFromTheme} from './generateTokensFromTheme';
export {Theme};

// utils
export {remToPx} from './utils/remToPx';
