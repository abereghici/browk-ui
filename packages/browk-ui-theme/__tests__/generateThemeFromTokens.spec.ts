import {generateThemeFromTokens} from '../src/generateThemeFromTokens';

describe('generateThemeFromTokens', () => {
  it('should generate a theme in the desired shape based on design tokens provided', () => {
    expect(
      generateThemeFromTokens({
        backgroundColors: {colorBackground: 'red'},
        borderColors: {colorBorder: 'blue'},
        borderWidths: {borderWidth0: '10px'},
        radii: {borderRadius0: '20px'},
        fonts: {fontFamilyCode: 'arial'},
        fontSizes: {fontSize80: '300px'},
        fontWeights: {fontWeightBold: '800'},
        lineHeights: {lineHeight0: '20px'},
        boxShadows: {shadow: 'grey'},
        sizings: {
          size0: '1px',
          size10: '1px',
          size20: '1px',
          size30: '1px',
          size40: '1px',
          size50: '1px',
          size60: '1px',
          size70: '1px',
          size80: '1px',
          size90: '1px',
          size100: '1px',
          size110: '1px',
          size120: '1px',
          size130: '1px',
          sizeIcon10: '1px',
          sizeIcon20: '1px',
          sizeIcon30: '1px',
          sizeIcon40: '1px',
          sizeIcon50: '1px',
          sizeIcon60: '1px',
          sizeIcon70: '1px',
          sizeIcon80: '1px',
        },
        spacings: {space10: '28px'},
        textColors: {colorText: 'black'},
        zIndices: {zIndex0: '-20'},
      })
    ).toMatchSnapshot();
  });
});
