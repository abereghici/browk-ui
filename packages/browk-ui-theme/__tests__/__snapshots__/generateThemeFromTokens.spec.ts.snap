// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateThemeFromTokens should generate a theme in the desired shape based on design tokens provided 1`] = `
Object {
  "backgroundColors": Object {
    "colorBackground": "red",
  },
  "borderColors": Object {
    "colorBorder": "blue",
  },
  "borderWidths": Object {
    "borderWidth0": "10px",
  },
  "breakpoints": Array [
    "1px",
    "1px",
    "1px",
  ],
  "fontSizes": Object {
    "fontSize80": "300px",
  },
  "fontWeights": Object {
    "fontWeightBold": "800",
  },
  "fonts": Object {
    "fontFamilyCode": "arial",
  },
  "heights": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "iconSizes": Object {
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "lineHeights": Object {
    "lineHeight0": "20px",
  },
  "maxHeights": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "maxWidths": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "minHeights": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "minWidths": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "radii": Object {
    "borderRadius0": "20px",
  },
  "shadows": Object {
    "shadow": "grey",
  },
  "sizes": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "space": Object {
    "space10": "28px",
  },
  "textColors": Object {
    "colorText": "black",
  },
  "widths": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size130": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
  },
  "zIndices": Object {
    "zIndex0": "-20",
  },
}
`;
